app/models/marc_fields/marc_field_wrapper.rb

```diff
@@ -0,0 +1,49 @@
+##
+# This class is intended to wrap Marc::DataFields that come from the marc gem.
+# This wrapper class is used to enhance the base Marc::DataField with additional behavior
+# such as parsing and retaining encoded data that might otherwise be removed for display (e.g. $6)
+# as well as add an attribute writer so subfields can be set as well as read.
```

Comment:
And we also use the `subfields` attribute writer in all classes to safely reject undesired subfields without risking removing them from other classes that may require them.